portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/AdvancedObjectAPI/ExtendPropertiesForSystemObjects.testcase

```diff
@@ -0,0 +1,112 @@
+@component-name = "portal-headless-frontend-infrastructure"
+definition {
+
+	property custom.properties = "feature.flag.LPS-135404=true";
+	property portal.release = "true";
+	property portal.upstream = "true";
+	property testray.main.component.name = "Headless Frontend Infrastructure";
+
+	setUp {
+		TestCase.setUpPortalInstanceNoSelenium();
+
+		task ("Given system User object with a new field added through the objects framework") {
+			var userObjectId = ObjectDefinitionAPI.getObjectDefinitionIdByName(name = "User");
+
+			ObjectFieldAPI.setUpGlobalObjectFieldId(
+				dbType = "String",
+				name = "passportNumber",
+				objectDefinitionId = "${userObjectId}");
+		}
+	}
```

Comment:
fyi @timpak, this is what we're depending on for Objects on the headless test. Just tagging you so you can see what we're doing and if you have any opinions/tips/questions to share on Objects testing side.